pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -220,16 +252,116 @@ func (h *handler) OnRemove(_ string, obj runtime.Object) (runtime.Object, error)
+			if apierror.IsNotFound(err) {
+				// we can't find the corresponding cluster, so we can go ahead and simply allow removal of the node.
+				logrus.Debugf("[MachineProvision] Proceeding with removal of dynamic machine %s as cluster was not found.", key)
+				return h.doRemove(obj)
+			} else if err != nil {
+				return obj, err
+			}
+			if !cluster.DeletionTimestamp.IsZero() {
+				// If the cluster deletion timestamp has been set, we can blindly proceed with delete.
+				logrus.Debugf("[MachineProvision] Cluster (%s) deletion timestamp was not zero that owns dynamic machine %s. Proceeding with delete", clusterName, key)
+				return h.doRemove(obj)
+			}
+			restConfig, err := h.kubeconfigManager.GetRESTConfig(cluster, cluster.Status)
+			if err != nil {
+				return obj, err
+			}
+
+			clientset, err := kubernetes.NewForConfig(restConfig)
+			if err != nil {
+				return obj, err
+			}
+
+			logrus.Debugf("[MachineProvision] Built K8s clientset to cluster: %s while processing dynamic machine removal %s", clusterName, key)
+
+			removeAnnotation := "etcd." + v2provruntime.GetRuntimeCommand(cluster.Spec.KubernetesVersion) + ".cattle.io/remove"
+			removedNodeNameAnnotation := "etcd." + v2provruntime.GetRuntimeCommand(cluster.Spec.KubernetesVersion) + ".cattle.io/removed-node-name"
+
+			machine, err := h.getMachine(obj)
+
+			if err != nil {
+				return obj, err
+			}
+
+			if machine.Status.NodeRef == nil {
+				// Machine noderef is nil, we should just allow deletion.
+				logrus.Debugf("[MachineProvision] There was no associated K8s node with this etcd dynamicmachine %s. proceeding with deletion", key)
+				return h.doRemove(obj)
+			}
+
+			logrus.Debugf("[MachineProvision] Retrieving node %s from K8s for the dynamic machine %s", machine.Status.NodeRef.Name, key)
+
+			node, err := clientset.CoreV1().Nodes().Get(context.TODO(), machine.Status.NodeRef.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierror.IsNotFound(err) {
+					logrus.Debugf("[MachineProvision] Node %s for dynamic machine %s was not found. proceeding with deletion", machine.Status.NodeRef.Name, key)
+					return h.doRemove(obj)
+				}
+				return obj, err
+			}
+
+			if val, ok := node.Annotations[removeAnnotation]; ok {
+				// check val to see if it's true, if not, continue
+				if val == "true" {
+					// check the status of the removal
+					logrus.Debugf("[MachineProvision] etcd member removal is currently in progress per the annotation %s", removeAnnotation)
+					if removedNodeName, ok := node.Annotations[removedNodeNameAnnotation]; ok {
+						// There is the possibility the annotation is defined, but empty.
+						if removedNodeName != "" {
+							logrus.Infof("[MachineProvision] Encountered empty value in removed node name annotation on dynamic machine %s, proceeding with deletion", key)
+							return h.doRemove(obj)
+						}
+					}
+					return obj, fmt.Errorf("remove annotation already set, waiting for node removal to be successful")
```

Comment:
If we are waiting for something to happen, I think a `generic.ErrSkip` should be return so we don't unnecessarily hammer the downstream cluster with GET node requests. If we need to reenqueue to reevaluate the infrastructure object, then reenqueuing every couple of seconds is probably better.